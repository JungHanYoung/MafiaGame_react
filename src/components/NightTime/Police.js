import React from 'react';
import { useGame } from '../../context/GameContext';

class Police extends React.Component {
	componentWillMount() {
		const { players, setNightOrder } = this.props;
		if (players.filter((person) => person.jobName === 'POLICE').length <= 0) {
			setNightOrder('result');
		}
	}
	detectingMafiaByPolice = (name) => {
		const { players } = this.props;
		players.find((person) => {
			return person.name === name;
		}).jobName === 'MAFIA'
			? this.setState({
					selected: name,
					isMafia: true
				})
			: this.setState({
					selected: name,
					isMafia: false
				});
	};
	render() {
		const { isMafia, selected } = this.state;
		const { players, setNightOrder } = this.props;
		return (
			<div>
				경찰의 차례입니다. 경찰은 마피아로 의심되는 사람을 지목해 마피아가 맞는지 확인할 수 있습니다.
				{players
					.filter((person) => {
						return person.jobName !== 'POLICE';
					})
					.map((person) => (
						<button onClick={() => this.detectingMafiaByPolice(person.name)}>{person.name}</button>
					))}
				{isMafia !== undefined && (
					<div>
						{selected}은 마피아가 {isMafia ? '맞습니다.' : '아닙니다.'}
						<button onClick={() => setNightOrder('result')}>밤 투표 결과</button>
					</div>
				)}
			</div>
		);
	}
}

export default useGame(({ state, actions }) => ({
	players: state.players,
	setNightOrder: actions.setNightOrder
}))(Police);
