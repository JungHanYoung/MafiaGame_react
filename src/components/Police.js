import React from 'react';

class Police extends React.Component {
	render() {
		return <div>Police</div>;
	}
}

export default Police;
