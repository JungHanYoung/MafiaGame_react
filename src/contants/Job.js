export const JOB_NAME_OF_MAFIA = 'MAFIA';
export const JOB_NAME_OF_POLICE = 'POLICE';
export const JOB_NAME_OF_DOCTOR = 'DOCTOR';
export const JOB_NAME_OF_CITIZEN = 'CITIZEN';
