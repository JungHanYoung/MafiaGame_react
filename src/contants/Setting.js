export const SELECT_OF_PEOPLE_NUM_START = 4;
export const SELECT_OF_PEOPLE_NUM_END = 13;
