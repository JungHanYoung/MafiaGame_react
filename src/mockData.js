import { List, Map } from "immutable";
import { JOB_NAME_OF_CITIZEN, JOB_NAME_OF_POLICE, JOB_NAME_OF_DOCTOR, JOB_NAME_OF_MAFIA } from "./contants/Job";


export const players = List([
    Map({
        name: "한영",
        jobName: JOB_NAME_OF_CITIZEN,
        daytimeVoted: 3
    }),
    Map({
        name: "병욱",
        jobName: JOB_NAME_OF_POLICE,
        daytimeVoted: 0
    }),
    Map({
        name: "지은",
        jobName: JOB_NAME_OF_DOCTOR,
        daytimeVoted: 0
    }),
    Map({
        name: "대용",
        jobName: JOB_NAME_OF_MAFIA,
        daytimeVoted: 0
    })
])

const playerWithoutImmutable = [
    {
        name: '한영',
        jobName: JOB_NAME_OF_CITIZEN
    },
    {
        name: '병욱',
        jobName: JOB_NAME_OF_MAFIA
    },
    {
        name: '대용',
        jobName: JOB_NAME_OF_DOCTOR
    },
    {
        name: '기수',
        jobName: JOB_NAME_OF_POLICE
    },
]

export const minPropsByGame = {
    history: {
        location: {
            state: {
                players: playerWithoutImmutable
            }
        }
    }
}

export const playersWithMafia = List([
    Map({
        name: "한영",
        jobName: JOB_NAME_OF_CITIZEN,
        daytimeVoted: 0
    }),
    Map({
        name: "병욱",
        jobName: JOB_NAME_OF_POLICE,
        daytimeVoted: 0
    }),
    Map({
        name: "지은",
        jobName: JOB_NAME_OF_DOCTOR,
        daytimeVoted: 0
    }),
    Map({
        name: "대용",
        jobName: JOB_NAME_OF_MAFIA,
        daytimeVoted: 0
    }),
    Map({
        name: "민기",
        jobName: JOB_NAME_OF_MAFIA,
        daytimeVoted: 0
    }),
    Map({
        name: "하림",
        jobName: JOB_NAME_OF_MAFIA,
        daytimeVoted: 0
    }),
])