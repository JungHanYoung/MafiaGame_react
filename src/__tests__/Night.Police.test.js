import React from 'react';
import { shallow } from 'enzyme'
import { List, Map } from 'immutable'
import Police from '../components/NightTime/Police'

import { players } from "../mockData";
import { JOB_NAME_OF_POLICE, JOB_NAME_OF_CITIZEN } from '../contants/Job';


const minProps = {
    players,
    me: players.find(player => player.get('jobName') === JOB_NAME_OF_POLICE),
    revoted: false,
    toggleConfirmed: jest.fn(),
    changeNightTimeOrder: jest.fn()
}

describe('Night > Police 컴포넌트 테스트', () => {

    it('렌더링 테스트', () => {
        const wrapper = shallow(<Police {...minProps} />)
        expect(wrapper.exists()).toBe(true)
    })

    it('스냅샷', () => {
        const wrapper = shallow(<Police {...minProps} />)
        expect(wrapper).toMatchSnapshot();
    })

    it('직업 확인 후 렌더링', () => {
        const wrapper = shallow(<Police {...minProps} />)

        wrapper.setState({
            selected: true,
            selectName: '한영',
            jobName: JOB_NAME_OF_CITIZEN
        })

        expect(wrapper).toMatchSnapshot()
    })

})