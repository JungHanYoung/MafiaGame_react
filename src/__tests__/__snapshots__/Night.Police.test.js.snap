// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Night > Police 컴포넌트 테스트 스냅샷 1`] = `
<Fragment>
  <Layout
    describe="경찰의 차례입니다.
							한 사람을 선택해 마피아가 맞는지
							확인할 수 있습니다."
  >
    <div
      className="vote-btn-container"
    >
      <div>
        <button
          className="btn-sm"
          key="police-select-0"
          onClick={[Function]}
        >
          한영
        </button>
        <button
          className="btn-sm"
          key="police-select-1"
          onClick={[Function]}
        >
          지은
        </button>
        <button
          className="btn-sm"
          key="police-select-2"
          onClick={[Function]}
        >
          대용
        </button>
      </div>
    </div>
  </Layout>
</Fragment>
`;

exports[`Night > Police 컴포넌트 테스트 직업 확인 후 렌더링 1`] = `
<Fragment>
  <Layout
    describe="경찰의 차례입니다.
							한 사람을 선택해 마피아가 맞는지
							확인할 수 있습니다."
  >
    <div>
      한영
      님은 
      마피아가 아닙니다.
    </div>
  </Layout>
  <button
    className="btn-lg"
    onClick={[Function]}
  >
    다음
  </button>
</Fragment>
`;
